src/stream-tcp-reassemble.c

```diff
@@ -934,9 +939,9 @@ static inline bool CheckGap(TcpStream *stream, Packet *p)
 static int ReassembleUpdateAppLayer (ThreadVars *tv,
         TcpReassemblyThreadCtx *ra_ctx,
         TcpSession *ssn, TcpStream *stream,
-        Packet *p, enum StreamUpdateDir dir)
+        Packet *p, enum StreamUpdateDir dir, uint32_t gap)
 {
-    const uint64_t app_progress = STREAM_APP_PROGRESS(stream);
+    const uint64_t app_progress = STREAM_APP_PROGRESS(stream) + gap;
```

Comment:
Ok, I'll have to rebase against your changes then make this change.

Currently I am never able to get the gap check below this to trigger. It was also the gap check before this which triggered, so the gap had to be used here to get at the next data (as the gap call had already been made to the app-layer before calling ReassembleUpdateAppLayer).